File: 프로그래머스/Lv2/올바른 괄호.cpp
#include<string>
#include <stack>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    stack<int> S;

    for(int i=0;i<s.size();i++){
        if(s[i]=='('){
            S.push(1);
        }else{
            if(!S.empty()){
                S.pop();
            }else{
                return false;
            }
        }
    }
    
    if(!S.empty()) answer=false;

    return answer;
}

File: 프로그래머스/Lv2/올바른 괄호.py
def solution(s):
    answer = True
    arr=list(s[0])
    
    for str in s[1:]:
        
        if(len(arr)==0): arr.append(str)
        elif(str=='('): arr.append(str)
        elif(str==')' and arr[-1]=='('): arr.pop()
        
    return len(arr)==0
